roles/openstack/nova/tasks/cleanup.yml

```diff
@@ -0,0 +1,45 @@
+---
+# This file is used to cleanup OpenStack nova resources
+
+- name: Register nova servers details
+  os_server_facts:
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+      server: "{{ prefix }}*"
+  register: servers_details
+- debug: var=servers_details
+
+- name: Set floating IP address
+  set_fact:
+      floating_ip_address: "{{ item.value[0].interface_ip }}"
+  with_dict: servers_details.ansible_facts
+  ignore_errors: yes
```

Comment:
"I am unsure if we should ignore_errors here?" I don't think we can assume every node has '"{{ item.value[0].interface_ip }}"'

"What happens if the floating IP is not set, will the Release floating IP work?" it will simply be skipped.

"Also this is used only in Release floating IPs, so why there is separate action to get just the interface_ip?" not sure I'm following. how do you suggest to retrieve this value without separate task?

"Isn't it easier to move this to the Release floating IPs part and remove whole section?"
I'm open to suggestions. Please post the code you think should replace this implementation.